crates/gpui/src/elements/animation.rs

```diff
@@ -41,6 +44,55 @@ impl Animation {
     }
 }
 
+/// An animation which can be applied to an element when transitioning between states
+pub struct TransitionAnimation {
+    /// The amount of time this animation should run for when transitioning from false to true.
+    /// When `None`, this transition isn't animated.
+    pub forward_duration: Option<Duration>,
```

Comment:
I don't think I like this API as written. The type name leads me to think that this is going to be a generalized "any kind of transition" animation, but the specific fields only apply to a boolean, and only on specific transitions of the boolean. Except for easing which has to be the same forward and backward.

We should figure out a more generalizable API for this, that we can implement for `bool`